bitchat/Features/media/ImageUtils.swift

```diff
@@ -0,0 +1,167 @@
+import Foundation
+#if os(iOS)
+import UIKit
+#else
+import AppKit
+import ImageIO
+import UniformTypeIdentifiers
+#endif
+
+enum ImageUtilsError: Error {
+    case invalidImage
+    case encodingFailed
+}
+
+enum ImageUtils {
+    private static let compressionQuality: CGFloat = 0.85
+    private static let targetImageBytes: Int = 60_000
+
+    static func processImage(at url: URL, maxDimension: CGFloat = 512) throws -> URL {
+        // Security H1: Check file size BEFORE reading into memory
+        let attrs = try FileManager.default.attributesOfItem(atPath: url.path)
+        guard let fileSize = attrs[.size] as? Int else {
+            throw ImageUtilsError.invalidImage
+        }
+        // Allow up to 10MB source images (will be scaled down)
+        guard fileSize <= 10 * 1024 * 1024 else {
+            throw ImageUtilsError.invalidImage
```

Comment:
I'd suggest adding "image is too large" to make the error message more meaningful to the user